pkg/network/server.go

```diff
@@ -1545,34 +1564,54 @@ func (s *Server) tryInitStateSync() {
 	if s.stateSync.IsInitialized() {
 		return
 	}
-
-	s.lock.RLock()
-	heights := make([]uint32, 0, len(s.peers))
-	for p := range s.peers {
-		if p.Handshaked() {
-			heights = append(heights, p.LastBlockIndex())
+	if !s.ServerConfig.NeoFSStateFetcherCfg.Enabled || s.syncStateFetcher.IsShutdown() {
+		s.lock.RLock()
+		heights := make([]uint32, 0, len(s.peers))
+		for p := range s.peers {
+			if p.Handshaked() {
+				heights = append(heights, p.LastBlockIndex())
+			}
 		}
-	}
-	s.lock.RUnlock()
-	slices.Sort(heights)
-	if len(heights) >= s.MinPeers && len(heights) > 0 {
-		// choose the height of the median peer as the current chain's height
-		h := heights[len(heights)/2]
-		err := s.stateSync.Init(h)
+		s.lock.RUnlock()
+		slices.Sort(heights)
+		if len(heights) >= s.MinPeers && len(heights) > 0 {
+			// choose the height of the median peer as the current chain's height
+			h := heights[len(heights)/2]
+			err := s.stateSync.Init(h)
+			if err != nil {
+				s.log.Fatal("failed to init state sync module",
+					zap.Uint32("evaluated chain's blockHeight", h),
+					zap.Uint32("blockHeight", s.chain.BlockHeight()),
+					zap.Uint32("headerHeight", s.chain.HeaderHeight()),
+					zap.Error(err))
+			}
+		}
+	} else {
+		s.log.Info("state sync is initializing based on the NeoFS container")
```

Comment:
All logs are expected to be symmetric. Add similar log to peer-based synchronization.